src/pages/ReportsUpload/index.tsx

```diff
@@ -0,0 +1,341 @@
+                    <EmptyStateIcon icon={ VolumeIcon } />
+                    <Title headingLevel={ TitleLevel.h5 } size="lg">
+                        Upload
+                    </Title>
+                    <div className="pf-c-empty-state__body">
+                        <Progress
+                            value={ this.props.progress }
+                            measureLocation={ ProgressMeasureLocation.outside }
+                            variant={ this.props.error ? ProgressVariant.danger : ProgressVariant.info }
+                        />
+                    </div>
+                    <EmptyStateBody>
+                        {
+                            this.props.success ?
+                                'Finished successfully. We will redirect you to the next page.' :
+                                'Your file is been uploaded, the process can take some time.'
+                        }
+                    </EmptyStateBody>
+                    <EmptyStateSecondaryActions>
+                        {
+                            this.props.success ? this.actionsOnUploadSuccess() : ''
+                        }
+                    </EmptyStateSecondaryActions>
+                </EmptyState>
+            </Bullseye>
+        );
+    }
+
+    form() {
+        return (
+            <Formik
+                initialValues={ this.initialFormValue }
+                validate={  this.validateForm }
+                onSubmit={ this.handleFormSubmit }
+            >
+                {
+                    props => <UploadForm
+                        onFileSelected={ this.onFileSelected }
+                        file={ this.props.file }
+                        handleCancel={ this.handleCloseModel }
+                        { ...props } />
+                }
+            </Formik>
+        );
+    }
+
+    render() {
+        return (
+            <React.Fragment>
+                <PageHeader>
+                    <PageHeaderTitle title='Reports' />
+                </PageHeader>
+                <Main>
+                    <Modal
+                        title='Report options'
+                        isLarge
+                        isOpen={ true }
+                        onClose={ this.handleCloseModel }
+                        ariaDescribedById="Report options"
+                        actions={ [] }
+                    >
+                        { this.state.showForm ? this.form() : this.progress() }
+                    </Modal>
+                </Main>
+            </React.Fragment>
+        );
+    }
+}
+
+const mapStateToProps = (state: GlobalState)  => {
+    let {
+        uploadState: {
+            file,
+            success,
+            error,
+            progress,
+            uploading
+        }
+    } = state;
+    return {
+        file,
+        success,
+        error,
+        progress,
+        uploading
+    };
+};
+
+const mapDispatchToProps = (dispatch: any) =>
```

Comment:
Unless you're doing something fancy with `mapDispatchToProps`, I don't think you need it. If all you're doing is binding some action creators you can just pass that object directly as the second argument of `connect()` instead of passing `mapDispatchToProps` there, and you don't need to call `bindActionCreators`. https://daveceddia.com/redux-mapdispatchtoprops-object-form/

Reply:
You're right. Thanks for the link and for the suggestion. I've just removed `bindActionCreators` and start using simple objects